stl/inc/functional

```diff
@@ -729,6 +730,50 @@ public:
 
 [[noreturn]] _CRTIMP2_PURE void __CLRCALL_PURE_OR_CDECL _Xbad_function_call();
 
+template <class _Rx, bool = is_void_v<_Rx>>
+struct _Invoker_ret {}; // helper to give INVOKE an explicit return type; avoids undesirable Expression SFINAE
+
+struct _Unforced { // tag to distinguish bind() from bind<R>()
+    explicit _Unforced() = default;
+};

```

Comment:
Style: should we move `_Unforced` up and make the partial specialization below be the primary `_Invoker_ret` template? When we have `true`/`false` specialization switching like this we conventionally make the `true` case the primary. It's notably peculiar that the primary template is defined but never used since every set of parameters matches one of the specializations below.

Reply:
Great catch - I believe this issue existed in the original code because when I implemented the `<functional>` overhaul with `_Invoker_ret` many years ago, we hadn't settled on the "prefer to use the primary template for `true`" convention. I agree that eliminating the unused definition will be a bonus clarity improvement.